subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -169,6 +152,24 @@ public GradleInternal getRoot() {
         }
     }
 
+    @Override
+    public void executeRootProjectActions() {
+        if (!isBaseProjectClassLoaderLocked()) {
```

Comment:
❓ Should we guard against this method (executeRootProjectActions) multiple calls?